py/nlrx64.c

```diff
@@ -35,8 +35,27 @@
 
 __attribute__((used)) unsigned int nlr_push_tail(nlr_buf_t *nlr);
 
+#if !MICROPY_NLR_OS_WINDOWS
+#if defined(__clang__) || (defined(__GNUC__) && __GNUC__ >= 8)
+#define USE_NAKED 1
+#else
+// On older gcc the equivalent here is to force omit-frame-pointer
+__attribute__((optimize("omit-frame-pointer")))
+#endif
+#endif
+
+#if !defined(USE_NAKED)
+#define USE_NAKED 0
+#endif
+
+#if USE_NAKED
+// nlr_push prelude should never push frame pointer register ebp onto the stack
+__attribute__((naked))
+#endif
 unsigned int nlr_push(nlr_buf_t *nlr) {
+    #if !USE_NAKED
```

Comment:
Is this guard needed?  The x86 version doesn't have it.

The argument is never used, so doesn't hurt to always have this void statement.

OTOH, I'm guessing that naked functions should not have any code in them except asm statements.

Reply:
The guard's needed here for Clang, it's more strict about the "nothing but asm statements" rule than gcc. (https://gcc.godbolt.org/z/fWjTT34aY)

nlrx86.c doesn't set `USE_NAKED` for Clang, even though I think it could (as per inline comment above.)

Reply:
(I can update nlrx86.c in this PR as well if you like. My only real concern would be that x86 probably has a slightly older range of compiler versions targeting it.)

Reply:
OK, thanks for the clarification.

Let's leave x86 as-is.